wire-ios-sync-engine/Source/UserSession/UserSession.swift

```diff
@@ -254,6 +254,10 @@ public protocol UserSession: AnyObject {
 
     var lastE2EIUpdateDateRepository: LastE2EIdentityUpdateDateRepositoryInterface? { get }
 
+    var fetchShareableConversationsUseCase: FetchShareableConversationsUseCaseProtocol { get }
+
+    var shareFileUseCase: ShareFileUseCaseProtocol { get }
```

Comment:
As described in the other comment, please let's reduce the user session as provider for everything and instantiate the use cases in the places they are used.

Reply:
done https://github.com/wireapp/wire-ios/pull/1658/commits/8c85d9ddbe3d4f96ae06fb91a9db48865e963819